dymos/phase/phase.py

```diff
@@ -1707,6 +1707,64 @@ def set_time_options(self, units=_unspecified, fix_initial=_unspecified,
         if name is not _unspecified:
             self.time_options['name'] = name
 
+    def add_duration_balance(self, name, val=0.0, index=None, units=None, mult_val=None, normalize=False):
```

Comment:
The word `add` here kind of implies that you can have many of these, but we only support a single balance residual.  I'd prefer that this was `set_duration_balance`.